diff --git a/libarchive/test/test_sparse_basic.c b/libarchive/test/test_sparse_basic.c
index 93710cb6..faa20abf 100644
--- a/libarchive/test/test_sparse_basic.c
+++ b/libarchive/test/test_sparse_basic.c
@@ -506,6 +506,7 @@ DEFINE_TEST(test_sparse_basic)
 	char *cwd;
 	struct archive *a;
 	const char *skip_sparse_tests;
+	char buf[PATH_MAX];
 	/*
 	 * The alignment of the hole of sparse files deeply depends
 	 * on filesystem. In my experience, sparse_file2 test with
@@ -584,7 +585,11 @@ DEFINE_TEST(test_sparse_basic)
 	/* Check if the filesystem where CWD on can
 	 * report the number of the holes of a sparse file. */
 #if defined(PATH_MAX) && !defined(__GLIBC__)
+#ifdef __MVS__
+	cwd = getcwd(buf, PATH_MAX);
+#else
 	cwd = getcwd(NULL, PATH_MAX);/* Solaris getcwd needs the size. */
+#endif
 #else
 	cwd = getcwd(NULL, 0);
 #endif
@@ -658,6 +663,7 @@ DEFINE_TEST(test_fully_sparse_files)
 	char *cwd;
 	struct archive *a;
 	const char *skip_sparse_tests;
+	char buf[PATH_MAX];
 
 	const struct sparse sparse_file[] = {
 		{ HOLE, MIN_HOLE }, { END, 0 }
@@ -673,7 +679,11 @@ DEFINE_TEST(test_fully_sparse_files)
 	/* Check if the filesystem where CWD on can
 	 * report the number of the holes of a sparse file. */
 #if defined(PATH_MAX) && !defined(__GLIBC__)
+#ifdef __MVS__
+	cwd = getcwd(buf, PATH_MAX);
+#else
 	cwd = getcwd(NULL, PATH_MAX);/* Solaris getcwd needs the size. */
+#endif
 #else
 	cwd = getcwd(NULL, 0);
 #endif
