diff --git a/unzip/bsdunzip.c b/unzip/bsdunzip.c
index f2bd1435..1bc09fbf 100644
--- a/unzip/bsdunzip.c
+++ b/unzip/bsdunzip.c
@@ -305,7 +305,11 @@ match_pattern(struct pattern_list *list, const char *str)
 
 	STAILQ_FOREACH(entry, list, link) {
 #ifdef HAVE_FNMATCH
-		if (fnmatch(entry->pattern, str, C_opt ? FNM_CASEFOLD : 0) == 0)
+#ifdef __MVS__
+	if (fnmatch(entry->pattern, str, C_opt ? 0 : 0) == 0)
+#else
+	if (fnmatch(entry->pattern, str, C_opt ? FNM_CASEFOLD : 0) == 0)
+#endif
 			return (1);
 #else
 #error "Unsupported platform: fnmatch() is required"
