diff --git a/libarchive/archive.h b/libarchive/archive.h
index d6de7fd1..f27a798a 100644
--- a/libarchive/archive.h
+++ b/libarchive/archive.h
@@ -174,7 +174,7 @@ __LA_DECL int		archive_version_number(void);
 /*
  * Textual name/version of the library, useful for version displays.
  */
-#define	ARCHIVE_VERSION_ONLY_STRING "3.8.0dev"
+#define	ARCHIVE_VERSION_ONLY_STRING "3.8.0"
 #define	ARCHIVE_VERSION_STRING "libarchive " ARCHIVE_VERSION_ONLY_STRING
 __LA_DECL const char *	archive_version_string(void);
 
