diff --git a/libarchive/archive_read_disk_posix.c b/libarchive/archive_read_disk_posix.c
index a7a98e9c..b5134f7e 100644
--- a/libarchive/archive_read_disk_posix.c
+++ b/libarchive/archive_read_disk_posix.c
@@ -1538,6 +1538,17 @@ set_statvfs_transfer_size(struct filesystem *fs, const struct statvfs *sfs)
 static inline __LA_UNUSED void
 set_statfs_transfer_size(struct filesystem *fs, const struct statfs *sfs)
 {
+#ifdef __MVS__
+	fs->xfer_align = -1;
+	fs->max_xfer_size = -1;
+#if defined(HAVE_STRUCT_STATFS_F_IOSIZE)
+	fs->min_xfer_size = sfs->f_iosize > 0 ? (long)sfs->f_iosize : -1;
+	fs->incr_xfer_size = sfs->f_iosize > 0 ? (long)sfs->f_iosize : -1;
+#else
+	fs->min_xfer_size = -1;
+	fs->incr_xfer_size = -1;
+#endif
+#else
 	fs->xfer_align = sfs->f_bsize > 0 ? (long)sfs->f_bsize : -1;
 	fs->max_xfer_size = -1;
 #if defined(HAVE_STRUCT_STATFS_F_IOSIZE)
@@ -1547,6 +1558,7 @@ set_statfs_transfer_size(struct filesystem *fs, const struct statfs *sfs)
 	fs->min_xfer_size = sfs->f_bsize > 0 ? (long)sfs->f_bsize : -1;
 	fs->incr_xfer_size = sfs->f_bsize > 0 ? (long)sfs->f_bsize : -1;
 #endif
+#endif
 }
 #endif
 
